api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptDao.java

```diff
@@ -0,0 +1,30 @@
+package org.pmiops.workbench.cdr.dao;
+import org.pmiops.workbench.cdr.model.Concept;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.data.repository.query.Param;
+
+import java.util.List;
+
+public interface ConceptDao extends CrudRepository<Concept, Long> {
+
+    @Query(value = "select c.* from cdr.concept c where c.concept_name like %:conceptName% and c.domain_id in ('Condition','Observation','Procedure', 'Measurement', 'Drug') order by c.count_value desc limit 10;",
+            nativeQuery = true)
+    List<Concept> findConceptLikeName(@Param("conceptName") String conceptName);
```

Comment:
LIKE on a fully loaded concept table is very slow. We should use full text indexes with MATCH instead, as shown here:

https://github.com/all-of-us/workbench/blob/78e630fae8642755c8d432be743ac7b9acdf3210/api/src/main/java/org/pmiops/workbench/cdr/dao/CriteriaDao.java#L17

https://github.com/all-of-us/workbench/blob/78e630fae8642755c8d432be743ac7b9acdf3210/api/db-cdr/changelog-schema/db.changelog-3.xml#L25

Reply:
Done.